pgmpy/inference/continuous_sampling.py

```diff
@@ -0,0 +1,579 @@
+        kinetic_change = 0.5 * np.float(np.dot(momentum_bar.T, momentum_bar) - np.dot(momentum.T, momentum))
+
+        acceptance_prob = np.exp(potential_change - kinetic_change)
+
+        # a = 2I[acceptance_prob] -1
+        a = 2 * (acceptance_prob > 0.5) - 1
+
+        condition = (acceptance_prob ** a) > (2 ** (-a))
+
+        while condition:
+            epsilon_app = (2 ** a) * epsilon_app
+
+            theta_bar, momentum_bar = self.discretize_time(self.grad_log_pdf, self.model,
+                                                           theta, momentum, epsilon_app).discretize_time()
+
+            _, logp = self.grad_log_pdf(theta, self.model).get_gradient_log_pdf()
+            _, logp_bar = self.grad_log_pdf(theta_bar, self.model).get_gradient_log_pdf()
+
+            potential_change = logp_bar - logp
+            kinetic_change = 0.5 * np.float(np.dot(momentum_bar.T, momentum_bar) - np.dot(momentum.T, momentum))
+
+            acceptance_prob = np.exp(potential_change - kinetic_change)
+
+            condition = (acceptance_prob ** a) > (2 ** (-a))
+
+        return epsilon_app
+
+    def sample(self, theta0, num_samples, Lambda, epsilon=None):
+        """
+        Method to return samples using Hamiltonian Monte Carlo
+
+        Parameters
+        ----------
+        theta0: A 1d array type object or a row matrix of shape 1 X d
+                or d X 1.(Will be converted to d X 1)
+                Vector representing values of parameter theta, the starting
+                state in markov chain.
+
+        num_samples: int
+                     Number of samples to be generated
+
+        Lambda: int or float
+                Target trajectory length, epsilon * number of steps(L),
+                where L is the number of steps taken per HMC iteration,
+                and epsilon is step size for splitting time method.
+
+        epsilon: float , defaults to None
+                 The step size for descrete time method
+                 If None, then will be choosen suitably
+
+
+        Returns
+        -------
+        list: A list of numpy array type objects containing samples
+
+        Examples
+        --------
+        >>> from pgmpy.inference import HamiltonianMC as HMC
+        >>> from pgmpy.inference import JointGaussianDistribution as JGD, GradLogPDFGaussian as GLPG
+        >>> from pgmpy.inference import LeapFrog
+        >>> import numpy as np
+        >>> mean = np.array([1, 1])
+        >>> cov_matrix = np.array([[1, 0.7], [0.7, 1]])
+        >>> model = JGD(mean, cov_matrix)
+        >>> sampler = HMC(model=model, grad_log_pdf=GLPG, discretize_time=LeapFrog)
+        >>> samples = sampler.sample(np.array([[1], [1]]), num_samples = 10000, Lambda=2, epsilon=None)
+        >>> samples_array = np.concatenate(samples, axis=1)
+        >>> np.cov(samples_array)
+        array([[ 0.64321553,  0.63513749],
+               [ 0.63513749,  0.98544953]])
+        """
+        # TODO: Proper parameterization
```

Comment:
@ankurankan This TODO needs to be resloved. Can you suggest how should I take parameter theta.